panic: file: not prefixed by //

## Describe the bug

`.chezmoiremove`:
```
etc/apt/sources.list.d/devel:kubic:libcontainers:stable.list*
```

`chezmoi.json.tmpl`
```
{
  "destDir": "/"
}
```

## To reproduce

Not sure yet if there is anything special in my environment, but:

```console
❯ ls -la /etc/apt/sources.list.d
total 40
drwxr-xr-x 2 root root 4096 Nov 21 11:43 .
drwxr-xr-x 7 root root 4096 Nov 20 19:54 ..
-rw-r--r-- 1 root root  100 Nov 20 19:54 devel:kubic:libcontainers:stable.list.save
-rw-r--r-- 1 root root  128 Nov 20 19:55 docker.list
-rw-r--r-- 1 root root   71 Nov 20 19:54 docker.list.save
-rw-r--r-- 1 root root  126 Nov 20 19:54 git-core-ubuntu-ppa-focal.list
-rw-r--r-- 1 root root  126 Nov 20 19:54 git-core-ubuntu-ppa-focal.list.save
-rw-r--r-- 1 root root  158 Nov 20 19:55 kubic.list
-rw-r--r-- 1 root root  203 Nov 20 19:54 vscode.list
-rw-r--r-- 1 root root  203 Nov 20 19:54 vscode.list.save
```

## Expected behavior

To remove all the files which starts with `etc/apt/sources.list.d/devel:kubic:libcontainers:stable.list`, like `etc/apt/sources.list.d/devel:kubic:libcontainers:stable.list.save`.

## Output of command with the `--verbose` flag

```console
$ sudo chezmoi apply --verbose --debug
2021-11-21T11:47:44-03:00 INF persistentPreRunRootE args=[] goVersion=go1.17.3 version={"builtBy":"goreleaser","commit":"10dbbcd2be5049cc4a5bfe97820188d17455c057","date":"2021-11-20T20:59:57Z","version":"2.8.0"}
2021-11-21T11:47:44-03:00 INF Stat component=system name=/home/felipecrs/.config/chezmoi/root.chezmoistate.boltdb
2021-11-21T11:47:44-03:00 ERR Stat error="stat /home/felipecrs/.dotfiles/root/.git: no such file or directory" component=system name=/home/felipecrs/.dotfiles/root/.git
2021-11-21T11:47:44-03:00 INF Stat component=system name=/home/felipecrs/.dotfiles/.git
2021-11-21T11:47:44-03:00 ERR ReadFile error="open /home/felipecrs/.dotfiles/root/.chezmoi.json.tmpl: no such file or directory" component=system data= name=/home/felipecrs/.dotfiles/root/.chezmoi.json.tmpl
2021-11-21T11:47:44-03:00 ERR ReadFile error="open /home/felipecrs/.dotfiles/root/.chezmoi.toml.tmpl: no such file or directory" component=system data= name=/home/felipecrs/.dotfiles/root/.chezmoi.toml.tmpl
2021-11-21T11:47:44-03:00 ERR ReadFile error="open /home/felipecrs/.dotfiles/root/.chezmoi.yaml.tmpl: no such file or directory" component=system data= name=/home/felipecrs/.dotfiles/root/.chezmoi.yaml.tmpl
2021-11-21T11:47:44-03:00 ERR ReadFile error="open /home/felipecrs/.dotfiles/root/.chezmoi.yml.tmpl: no such file or directory" component=system data= name=/home/felipecrs/.dotfiles/root/.chezmoi.yml.tmpl
2021-11-21T11:47:44-03:00 ERR ReadFile error="open /home/felipecrs/.dotfiles/root/.chezmoi.properties.tmpl: no such file or directory" component=system data= name=/home/felipecrs/.dotfiles/root/.chezmoi.properties.tmpl
2021-11-21T11:47:44-03:00 ERR ReadFile error="open /home/felipecrs/.dotfiles/root/.chezmoi.props.tmpl: no such file or directory" component=system data= name=/home/felipecrs/.dotfiles/root/.chezmoi.props.tmpl
2021-11-21T11:47:44-03:00 ERR ReadFile error="open /home/felipecrs/.dotfiles/root/.chezmoi.prop.tmpl: no such file or directory" component=system data= name=/home/felipecrs/.dotfiles/root/.chezmoi.prop.tmpl
2021-11-21T11:47:44-03:00 ERR ReadFile error="open /home/felipecrs/.dotfiles/root/.chezmoi.hcl.tmpl: no such file or directory" component=system data= name=/home/felipecrs/.dotfiles/root/.chezmoi.hcl.tmpl
2021-11-21T11:47:44-03:00 ERR ReadFile error="open /home/felipecrs/.dotfiles/root/.chezmoi.tfvars.tmpl: no such file or directory" component=system data= name=/home/felipecrs/.dotfiles/root/.chezmoi.tfvars.tmpl
2021-11-21T11:47:44-03:00 ERR ReadFile error="open /home/felipecrs/.dotfiles/root/.chezmoi.dotenv.tmpl: no such file or directory" component=system data= name=/home/felipecrs/.dotfiles/root/.chezmoi.dotenv.tmpl
2021-11-21T11:47:44-03:00 ERR ReadFile error="open /home/felipecrs/.dotfiles/root/.chezmoi.env.tmpl: no such file or directory" component=system data= name=/home/felipecrs/.dotfiles/root/.chezmoi.env.tmpl
2021-11-21T11:47:44-03:00 ERR ReadFile error="open /home/felipecrs/.dotfiles/root/.chezmoi.ini.tmpl: no such file or directory" component=system data= name=/home/felipecrs/.dotfiles/root/.chezmoi.ini.tmpl
2021-11-21T11:47:44-03:00 ERR Mkdir error="mkdir /home/felipecrs/.config/chezmoi: file exists" component=system name=/home/felipecrs/.config/chezmoi perm=511
2021-11-21T11:47:44-03:00 INF Stat component=system name=/home/felipecrs/.config/chezmoi
2021-11-21T11:47:44-03:00 INF Get bucket=configState component=persistentState key=configState value=
2021-11-21T11:47:44-03:00 ERR ReadFile error="open /home/felipecrs/.dotfiles/root/.chezmoiroot: no such file or directory" component=system data= name=/home/felipecrs/.dotfiles/root/.chezmoiroot
2021-11-21T11:47:44-03:00 INF Stat component=system name=/home/felipecrs/.dotfiles/root
2021-11-21T11:47:44-03:00 INF Stat component=system name=/home/felipecrs/.dotfiles/root
2021-11-21T11:47:44-03:00 INF ReadDir component=system name=/home/felipecrs/.dotfiles/root
2021-11-21T11:47:44-03:00 INF ReadFile component=system data="\"usr/share/keyrings/docker-archive-keyring.gpg\":\n  type: file\n  ..." name=/home/felipecrs/.dotfiles/root/.chezmoiexternal.yaml
2021-11-21T11:47:44-03:00 INF ReadFile component=system data="NAME=\"Ubuntu\"\nVERSION=\"20.04.3 LTS (Focal Fossa)\"\nID=ubuntu\nID_L..." name=/etc/os-release
2021-11-21T11:47:44-03:00 INF ReadFile component=system data="etc/apt/sources.list.d/devel:kubic:libcontainers:stable.list.*\n" name=/home/felipecrs/.dotfiles/root/.chezmoiremove
panic: /etc/apt/sources.list.d/devel:kubic:libcontainers:stable.list.save: not prefixed by //

goroutine 1 [running]:
github.com/twpayne/chezmoi/v2/internal/chezmoi.mustTrimPrefix({0xc0006054a0, 0x42}, {0xc0005b9a9c, 0x2})
        /home/runner/work/chezmoi/chezmoi/internal/chezmoi/chezmoi.go:169 +0x10a
github.com/twpayne/chezmoi/v2/internal/chezmoi.(*patternSet).glob(0xc0003dc5d0, {0x11ef1c8, 0x1ab0960}, {0xc0005b9a9c, 0x2})
        /home/runner/work/chezmoi/chezmoi/internal/chezmoi/patternset.go:75 +0x345
github.com/twpayne/chezmoi/v2/internal/chezmoi.(*SourceState).Read.func1({{0xc00050df80, 0x20}}, {0x11dc8e8, 0xc0000ac9c0}, {0x0, 0x0})
        /home/runner/work/chezmoi/chezmoi/internal/chezmoi/sourcestate.go:757 +0x4c8
github.com/twpayne/chezmoi/v2/internal/chezmoi.walkSourceDir({0x11ee5c0, 0xc0005a5d40}, {{0xc00050df80, 0x1}}, {0x11dc8e8, 0xc0000ac9c0}, 0xc0005f50f8)
        /home/runner/work/chezmoi/chezmoi/internal/chezmoi/system.go:187 +0x7d
github.com/twpayne/chezmoi/v2/internal/chezmoi.walkSourceDir({0x11ee5c0, 0xc0005a5d40}, {{0xc000580a60, 0x0}}, {0x11dc8e8, 0xc000661ad0}, 0xc0005f50f8)
        /home/runner/work/chezmoi/chezmoi/internal/chezmoi/system.go:227 +0x367
github.com/twpayne/chezmoi/v2/internal/chezmoi.WalkSourceDir({0x11ee5c0, 0xc0005a5d40}, {{0xc000580a60, 0xc000118030}}, 0xc0005f50f8)
        /home/runner/work/chezmoi/chezmoi/internal/chezmoi/system.go:167 +0x9b
github.com/twpayne/chezmoi/v2/internal/chezmoi.(*SourceState).Read(0xc0000a4280, {0x11d7938, 0xc000134000}, 0xc000118030)
        /home/runner/work/chezmoi/chezmoi/internal/chezmoi/sourcestate.go:818 +0x1b6
github.com/twpayne/chezmoi/v2/internal/cmd.(*Config).newSourceState(0xc00051f200, {0x11d7938, 0xc000134000}, {0x0, 0x0, 0xb})
        /home/runner/work/chezmoi/chezmoi/internal/cmd/config.go:1354 +0xc5c
github.com/twpayne/chezmoi/v2/internal/cmd.(*Config).applyArgs(0xc00051f200, {0x11d7938, 0xc000134000}, {0x11ee5c0, 0xc0005a5da0}, {{0x1a37958, 0xc00011b340}}, {0xc00007ea80, 0x0, 0x4}, ...)
        /home/runner/work/chezmoi/chezmoi/internal/cmd/config.go:545 +0x496
github.com/twpayne/chezmoi/v2/internal/cmd.(*Config).runApplyCmd(0x5a2280, 0xc0005f5c00, {0xc00007ea80, 0x0, 0x0})
        /home/runner/work/chezmoi/chezmoi/internal/cmd/applycmd.go:39 +0x150
github.com/spf13/cobra.(*Command).execute(0xc0005ac780, {0xc00007e940, 0x4, 0x4})
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.2.1/command.go:856 +0x60e
github.com/spf13/cobra.(*Command).ExecuteC(0xc0005ac000)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.2.1/command.go:974 +0x3bc
github.com/spf13/cobra.(*Command).Execute(...)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.2.1/command.go:902
github.com/twpayne/chezmoi/v2/internal/cmd.(*Config).execute(0xc0005f5e00, {0xc000124130, 0x5, 0x5})
        /home/runner/work/chezmoi/chezmoi/internal/cmd/config.go:1075 +0x78
github.com/twpayne/chezmoi/v2/internal/cmd.runMain({{0x11b3ce8, 0x5}, {0x11ceb40, 0x28}, {0x11b7570, 0x14}, {0x11b4b68, 0xa}}, {0xc000124130, 0x5, ...})
        /home/runner/work/chezmoi/chezmoi/internal/cmd/cmd.go:289 +0x151
github.com/twpayne/chezmoi/v2/internal/cmd.Main({{0x11b3ce8, 0x5}, {0x11ceb40, 0x28}, {0x11b7570, 0x14}, {0x11b4b68, 0xa}}, {0xc000124130, 0x5, ...})
        /home/runner/work/chezmoi/chezmoi/internal/cmd/cmd.go:100 +0xc9
main.main()
        /home/runner/work/chezmoi/chezmoi/main.go:26 +0x125
```

## Output of `chezmoi doctor`

```console
$ chezmoi doctor
```

## Additional context

If I do not use a star, it works normally.

--- a/internal/chezmoi/chezmoi.go
+++ b/internal/chezmoi/chezmoi.go
@@ -179,3 +179,11 @@ func mustTrimSuffix(s, suffix string) string {
 	}
 	return s[:len(s)-len(suffix)]
 }
+
+// ensureSuffix adds suffix to s if s is not suffixed by suffix.
+func ensureSuffix(s, suffix string) string {
+	if strings.HasSuffix(s, suffix) {
+		return s
+	}
+	return s + suffix
+}

--- a/internal/chezmoi/sourcestate.go
+++ b/internal/chezmoi/sourcestate.go
@@ -754,7 +754,7 @@ func (s *SourceState) Read(ctx context.Context, options *ReadOptions) error {
 			if err := s.addPatterns(removePatterns, sourceAbsPath, sourceRelPath); err != nil {
 				return err
 			}
-			matches, err := removePatterns.glob(s.system.UnderlyingFS(), s.destDirAbsPath.String()+"/")
+			matches, err := removePatterns.glob(s.system.UnderlyingFS(), ensureSuffix(s.destDirAbsPath.String(), "/"))
 			if err != nil {
 				return err
 			}

--- a/internal/cmd/testdata/scripts/remove.txt
+++ b/internal/cmd/testdata/scripts/remove.txt
@@ -29,8 +29,24 @@ chezmoi apply
 ! exists $HOME/.file
 ! exists $HOME/.dir
 
+chhome home3/user
+
+# test that chezmoi apply with .chezmoiremove with star works on destination dir with trailing slash
+exists $HOME/.star-file
+exists $HOME/.star-dir
+chezmoi apply --destination=$HOME/
+! exists $HOME/.star-file
+! exists $HOME/.star-dir
+
 -- home2/user/.dir/.keep --
 -- home2/user/.file --
 # contents of .file
 -- home2/user/.local/share/chezmoi/remove_dot_file --
 -- home2/user/.local/share/chezmoi/remove_dot_dir --
+
+-- home3/user/.star-dir/.keep --
+-- home3/user/.star-file --
+# contents of .star-file
+-- home3/user/.local/share/chezmoi/.chezmoiremove --
+.*-dir/
+.*-file

